### PR TITLE
Add retry to prevent network timeout

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [X] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Context

Adds retries for bundletool downloads to prevent timeout errors.

### CODE REVIEW

File: bundletool/bundletool.go [hunk 1]
```diff
@@ -1,71 +1,41 @@
...
 import (
-	"fmt"
-	"io"
-	"net/http"
-	"os"
 	"path/filepath"
 
 	"github.com/bitrise-io/go-utils/command"
 	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-steplib/bitrise-step-export-universal-apk/filedownloader"
```
Comment (mateherber) [inline bundletool/bundletool.go:8]: This is being moved right now :)

File: bundletool/retryhttp.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// RetryLogAdaptor adapts the retryablehttp.Logger interface to the go-utils logger.
+type RetryLogAdaptor struct{}
+
+// Printf implements the retryablehttp.Logger interface
+func (*RetryLogAdaptor) Printf(fmtStr string, vars ...interface{}) {
+	log.Debugf(fmtStr, vars...)
+}
+
+// NewRetryableClient returns a retryable HTTP client
+func NewRetryableClient() *retryablehttp.Client {
```
Comment (mateherber) [inline bundletool/retryhttp.go:17]: This should go to the go-utils too

Comment (mateherber) [reply bundletool/retryhttp.go:17]: What  is the default "backoff" strategy?

**Review by mateherber — CHANGES_REQUESTED** [review]

File: bundletool/retryhttp.go (hunk 2 above)
Comment (lpusok) [reply bundletool/retryhttp.go:17]: It is an exponential backoff (limited by min/max settings): https://github.com/hashicorp/go-retryablehttp/blob/02c1586c8f14be23e7eeb522f1094afbabf45e93/client.go#L472

**Review by mateherber — APPROVED** [review]

**Pull request merged**